src/components/select/CustomSelect.tsx

```diff
@@ -0,0 +1,259 @@
+
+  // If the dropdown is cut off at the bottom of the viewport
+  // adjust the height to fit within the viewport minus fixed margin.
+  // This usually becomes an issue when the dropdown is at the bottom of the viewport or screen getting smaller.
+  if (bottomToViewportBottom < 0) {
+    const adjustedHeight = dropdownHeight + bottomToViewportBottom - 20;
+    dropdown.style.height = `${adjustedHeight}px`;
+    dropdown.style.maxHeight = `${adjustedHeight}px`;
+    return;
+  }
+
+  // If the dropdown is taller than the max height, do not adjust.
+  if (dropdownHeight >= DROPDOWN_MAX_HEIGHT) {
+    return;
+  }
+
+  // If the dropdown does not have overflow, do not adjust.
+  const hasOverflow = dropdown.scrollHeight > dropdown.clientHeight;
+  if (!hasOverflow) {
+    return;
+  }
+
+  // If the dropdown is not cut off at the bottom of the viewport
+  // adjust the height of the dropdown so that its bottom edge is 20px from the bottom of the viewport.
+  const topToViewportBottom = window.innerHeight - dropdownRect.top;
+  const adjustedHeight = topToViewportBottom - 20;
+  dropdown.style.height = `${adjustedHeight}px`;
+  dropdown.style.maxHeight = `${adjustedHeight}px`;
+};
+
+const adjustDropdownHeightAbove = (dropdown: HTMLElement) => {
+  // The search height is subtracted (if necessary) so that no options will be hidden behind the search input.
+  const search = document.querySelector(
+    ".p-custom-select__search",
+  ) as HTMLElement;
+  const searchRect = search?.getBoundingClientRect();
+  const searchHeight = searchRect?.height || 0;
+  const dropdownRect = dropdown.getBoundingClientRect();
+  const bottomToViewportTop = dropdownRect.bottom;
+  const dropdownHeight = dropdown?.offsetHeight || 0;
+
+  // If the dropdown is taller than the top edge of the viewport
+  // adjust the height to fit within the viewport minus fixed margin.
+  if (dropdownHeight > bottomToViewportTop) {
+    const adjustedHeight = bottomToViewportTop - searchHeight - 20;
+    dropdown.style.height = `${adjustedHeight}px`;
+    dropdown.style.maxHeight = `${adjustedHeight}px`;
+    return;
+  }
+
+  // If the dropdown is taller than the max height, do not adjust.
+  if (dropdownHeight >= DROPDOWN_MAX_HEIGHT) {
+    return;
+  }
+
+  // If the dropdown does not have overflow, do not adjust.
+  const hasOverflow = dropdown.scrollHeight > dropdown.clientHeight;
+  if (!hasOverflow) {
+    return;
+  }
+
+  // If the dropdown is not cut off at the top of the viewport
+  // adjust the height of the dropdown so that its top edge is 20px from the top of the viewport.
+  const adjustedHeight = bottomToViewportTop - searchHeight - 20;
+  dropdown.style.height = `${adjustedHeight}px`;
+  dropdown.style.maxHeight = `${adjustedHeight}px`;
+};
+
+const adjustDropdownHeight = () => {
+  const dropdown = document.querySelector(
+    ".p-custom-select__dropdown .p-list",
+  ) as HTMLElement;
+  const dropdownRect = dropdown.getBoundingClientRect();
```

Comment:
I can't seem to recreate the error on the demo server or locally (using device view or just adjusting the height of the browser), I wonder if it's a browser version issue? Maybe you can show me how you did it?

I've updated the code to use refs as suggested, but only for the dropdown list and search input. The actual toggle is inside the `ContextualMenu` component which means an upstream change will be required for the `ContextualMenu` and `Button` components (since the ref must be attached to the button). Alternatively, I can pass a toggle `ReactNode` into `ContextualMenu`, but that's ugly as I would copy the button code and move it into the `CustomSelect` component. I think to limit the scope of this PR, a `querySelector` for the toggle element should be okay?

Reply:
Can't reproduce it any more either, seems resolved.